### ISSUE TITLE
Use `<RichText>` component for description fields

### ISSUE THREAD

**Issue by joshfeck** [eventespresso/event-espresso-core#1513]
Darren noted:

> Currently those fields in the legacy editor allow for html.  I think we can both preserve and enhance this by using the RichText api available from the Gutenberg project (which includes built in extensibility for formats etc).

> see https://github.com/WordPress/gutenberg/tree/master/packages/rich-text
> see https://github.com/WordPress/gutenberg/tree/master/packages/block-editor/src/components/rich-text

> There will be some potential overlap with the "InlineEdit" component already introduced in the EDTR refactor and might be some conflicts with usage with ReactFinalForm, but still doable I think.

**Comment by nicolad** [comment]
I tried to use `RichText` in two scenarios, but I am missing smth ATM.
In first case I tried to use it within EDTR, but got this error:

![image](https://user-images.githubusercontent.com/1477453/86135116-10ce5b80-baf3-11ea-8330-62b35700e7b6.png)

tried `RichText` from  both `@wordpress/rich-text` and `@wordpress/blocks`
***
in the second scenario, since there were some errors pointing to webpack, I tried to use it with a new and empty created CRA, but still got some errors

![image](https://user-images.githubusercontent.com/1477453/86135381-686cc700-baf3-11ea-8617-b8d4ea842d43.png)

***
will try to configure it later
***
also found these alternatives: https://www.npmtrends.com/draft-js-vs-quill-vs-slate
@tn3rb lemme know if you like smth from the above :point_up:  (in case I won't be able to configure it)
***
bottom line: I am not yet fully convinced, but I tend to think that `RichText` component is designed to be used only in context of GB blocks

**Comment by tn3rb** [comment]
> tried RichText from both @wordpress/rich-text and @wordpress/blocks

we definitely can't use the latter of those two. Would have to be the packages version:
https://github.com/WordPress/gutenberg/tree/master/packages/rich-text

If that can't be loaded and we have to go with one of the others, then draft and quill both look pretty good, but slate seem like it is missing some functionality that we might want.

#### Draft

- https://draftjs.org/ && https://github.com/facebook/draft-js
- looks the best (imho) as is
- appears to have some really advanced functionality and potentially be very powerful/flexible/etc 
- docs look way more complicated and not as easy to follow
- examples?


#### Quill

- https://quilljs.com/docs/quickstart/ && https://github.com/quilljs/quill
- not as elegant looking out of the box, but can be themed (ie: make it blend better with EE)
- appears to have everything we would need and more
- appears to be really clean, simple, and straightforward to use
- great documentation 
- good examples

Although Draft looks **really** powerful, Quill looks like it would be way easier to deal with and appears to be capable of doing anything we would want, so I would be inclined to use that, although I think we should take another stab at getting the WordPress RichText component working.

**Comment by nicolad** [comment]
> although I think we should take another stab at getting the WordPress RichText component working

for the record, I'll attach also the discussion I had with  a8c/ GB contributor:
![image](https://user-images.githubusercontent.com/1477453/86805086-3a9dfa00-c080-11ea-842a-41cdfc8d8806.png)
***
> I'll ask @manzoorwanijk to take a look at this as well

agree, @manzoorwanijk you can take a look also, maybe I missed smth when I tried to integrate that

**Comment by manzoorwanijk** [comment]
After having a look at all the above options, I think RichText is not yet ready to be used outside GB because it internally stores the editor state in a redux store inside `@wordpress/data`.

Regarding DraftJS, it looks great and it uses the React state to store the state of the editor and is fully customizable unless other options out there. Also it has a decent API.

Quill uses a DOM container to render the editor which is not a good option for us because we my need render it inside modals, inline editors and who knows where else?

IMHO, we should create a separate package for this in barista named `content-editor` or `rich-text` or something similar and then create a generic API that can be exposed for usage. This will be a safer option just in case we want to switch to something else in future.

**Comment by manzoorwanijk** [comment]
WP `RichText` API is tightly coupled with wp `data` and every utility seems to have been written thinking that it will always reside in GB.

**Comment by tn3rb** [comment]
OK let's go ahead with the following:

- create a new `RichTextEditor` package in barista

- add DraftJS as a dependency

- decouple the DraftJS from the the public API via a Facade layer

- be prepared to integrate directly in the existing EDTR prior to release
